src/Nancy/NancyEngine.cs

```diff
@@ -252,7 +253,24 @@ private void CheckStatusCodeHandler(NancyContext context)
                 return;
             }
 
-            handler.Handle(context.Response.StatusCode, context);
+            try
+            {
+                handler.Handle(context.Response.StatusCode, context);
+            }
+            catch (Exception ex)
+            {
+                context.Response = new Response()

```

Comment:
Instead of doing this, we should call `defaultHandler.Handle` so we get the beautiful Nancy error page instead of just a simple string.